reco_utils/evaluation/spark_evaluation.py

```diff
@@ -470,3 +479,457 @@ def _get_relevant_items_by_timestamp(
     )
 
     return items_for_user
+
+
+class SparkDiversityEvaluation:
```

Comment:
why is this class in the same module as other classes? imo we should be moving towards modules that have a single class in them instead of making larger modules with multiple classes.

Reply:
I think that whichever way we choose it has to be consistent i.e. either there are separate modules for rating, ranking and diversity, or one module containing all of them. We currently follow the latter approach, so in this PR I am making the code consistent with this. We could discuss whether to change approach, but that would mean reorganizing all the evaluation code and would go beyond the scope of this PR.

Reply:
ok fair enough, they're all related so it's not that bad, i think my gut reaction is just that the file is very long now. I'll put in a feature to capture discussion around this.

Reply:
I agrees with Scott. If we want to make the code consistent at this point, I would suggest to break down spark_evaluation.py into spark_rating_evaluation.py (SparkRatingEvaluation class) and spark_ranking_evaluation.py (SparkRankingEvaluation class).